src/dev/flang/ast/AstErrors.java

```diff
@@ -424,7 +424,12 @@ else if (frmlT.compareTo(Types.resolved.t_unit) == 0)
           }
         else
           {
-            remedy = "To solve this, you could change the type of the target " + ss(target) + " to " + s(actlT) + " or convert the type of the assigned value to " + s(frmlT) + ".\n";
+            remedy = !frmlT.isRef() && actlT.feature().inheritsFrom(frmlT.feature()) ?
+                        "To solve this you could:\n" + //
+                            "  • make  " + s(frmlT) + " a reference by adding the " + st("ref")+ " keyword, so all its heirs can be used in place of it\n" + //
+                            "  • change the type of the target " + ss(target) + " to " + s(actlT) + "\n" + //
+                            "  • convert the type of the assigned value to " + s(frmlT) + "."
```

Comment:
It is not clear if you have to do all three of these things or pick just one. Add `or` and commas for clarity:
```suggestion
                            "  • make  " + s(frmlT) + " a reference by adding the " + st("ref")+ " keyword, so all its heirs can be used in place of it,\n" + //
                            "  • change the type of the target " + ss(target) + " to " + s(actlT) + ", or\n" + //
                            "  • convert the type of the assigned value to " + s(frmlT) + "."
```